src/pages/student/profile/EditProfile/index.js

```diff
@@ -3,38 +3,211 @@ import Button from 'react-bootstrap/Button';
+    try {
+      await ProfileEditApi.profileEdit({ name, email, password });
+      window.location = '/profile/view';
+    } catch (error) {
+      console.log(error.response);
+      if (error?.response?.data?.errors) {
+        setErrors(error?.response?.data?.errors);
+      } else if (error?.response?.data?.error) {
+        showAlertDialog(true, 'The password you have entered is incorrect.');
+      } else {
+        showAlertDialog(true, 'An error has occurred.');
+      }
+    }
+  };
+
   return (
-    <div className="d-flex justify-content-center align-items-center text-align-center" style={{marginTop:'150px'}} >
-      <Card style={{ width: '430px', padding:'50px', paddingTop:'20px', backgroundColor:'#E0EAEC' }}>
+    <div
+      className="d-flex justify-content-center align-items-center text-align-center"
+      style={{ marginTop: '150px' }}
+    >
+      {showAlert && (
+        <Alert
+          className={style.allertstyle}
+          variant="danger"
+          onClose={() => setShowAlert(false)}
+          dismissible
+        >
+          {alertMessage}
+        </Alert>
+      )}
+      <Card
+        style={{
+          width: '430px',
+          padding: '50px',
+          paddingTop: '20px',
+          backgroundColor: '#E0EAEC',
+        }}
+      >
         <div className={style.HeadingText}>Edit Account Info</div>
-        <Form style={{marginTop:'20px'}}>
-          <Form.Group className="mb-3" controlId="formBasicEmail">
-            <Form.Label style={{ marginBottom: '0px', color: '#48535B', fontSize:'16px' }}>Name</Form.Label>
-            <Form.Control style={{fontSize:'14px' }} type="text" name="name" value="Jane Doe" />
-          </Form.Group>
-
-          <Form.Group className="mb-4" controlId="formBasicPassword">
-            <Form.Label style={{ marginBottom: '0px', color: '#48535B' }}>Email</Form.Label>
-            <Form.Control style={{fontSize:'14px' }} type="email" name="email" value="janedoe@gmail.com" />
-          </Form.Group>
-          <Button
-            className={style.changepassbutton}
-            variant="primary"
-            type="submit"
-          >
-            Change
-          </Button>
-          <div>
-            <a className={style.cancel} href="/profile">
-              Cancel
-            </a>
+        {profileName === null ? (
+          <div className={style.loading}>
+            <Spinner animation="border" role="status"></Spinner>
+            <span className={style.loadingWord}>Loading</span>
           </div>
-        </Form>
+        ) : (
+          <Form
+            onSubmit={handleSubmit(handleOnSubmit)}
+            style={{ marginTop: '20px' }}
+          >
+            <Form.Group className="mb-3" controlId="formBasicEmail">
+              <Form.Label
+                style={{
+                  marginBottom: '0px',
+                  color: '#48535B',
+                  fontSize: '16px',
+                }}
```

Comment:
please check if you can use a class here as well. this has the same style with lines 125 and 157. the difference is only the font-size